### PR TITLE
Create intel-hex.d.ts and add types field

### PR DESCRIPTION
**Pull request by cinderblock**
Incorporate types directly into package as manually maintained type declarations.

TypeScript (and other) users of this package will have automatically detectable types.

### CODE REVIEW

**Comment by FaustXVI** [comment]
That would be awesome !
However, the documentation of `slidePad` was incorrect (#22) and thus the typing here is also incorrect.

File: intel-hex.d.ts [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+   *
+   * <br/>
+   * This method allocates new memory.
+   *
+   * @param {Number} address The start address of the slice
+   * @param {Number} length The length of memory map to slice out
+   * @param {Number} [padByte=0xFF] The value of the byte assumed to be used as padding
+   * @return {MemoryMap}
+   */
+  slicePad(address: number, length: number, padByte?: number): MemoryMap;
```
Comment (FaustXVI) [inline intel-hex.d.ts]:
```suggestion
   * @return {Uint8Array}
   */
  slicePad(address: number, length: number, padByte?: number): Uint8Array;
```

Comment (cinderblock) [reply intel-hex.d.ts]: Yep. We should probably fix the function header in the source file to match this correct type.

**Comment by cinderblock** [comment]
~Oops. Latest change is redundant with #22.~ Edit: reverted

Might need to do a minor version bump to publish latest to Npm

**Review by datenreisender — COMMENTED** [review]
Thanks for adding types! 
(And please excuse that it took so long to take a look at it)

We also already added types in projects were we needed them: In [`programmer`](https://github.com/NordicSemiconductor/pc-nrfconnect-programmer/blob/36bd19253fdd7135d0348026babfa05b49f7ab9e/types/nrf-intel-hex/index.d.ts) and [`shared`](https://github.com/NordicSemiconductor/pc-nrfconnect-shared/blob/7ee0676230a8cd09427e03c6db95b781eef7f1be/typings/nrf-intel-hex.d.ts). But of course it makes a lot more sense to add the types here.

I will merge this PR as it is and then probably merge our own types into this (so that types like `MemoryBlocks` are exported). And hopefully I will release a version 1.4.0 of this soon.

**Review by datenreisender — APPROVED** [review]

**Comment by datenreisender** [comment]
@cinderblock @FaustXVI: Thanks again for providing these change and nudging us about it. 

A new version 1.4.0 of `nrf-intel-hex` which includes theses changes is now released to npm: [nrf-intel-hex](https://www.npmjs.com/package/nrf-intel-hex/v/1.4.0). 

I already switched to this version for our own projects and there it works just fine.

**Pull request merged**